### ISSUE TITLE
New ftp backend can't see inside folders containing square brackets

### ISSUE THREAD

**Issue by ghost** [rclone/rclone#1438]
I run a daily sync with an ftp server, for which I usually use curlftps for the ftp backend, I tried using the rclone ftp backend and seem to have discovered something peculiar. rclone doesn't seem to see any files in folders containing square brackets. This issue occurs both when I mount on windows and using rclone `check` and `sync` on linux.

**Comment by ncw** [comment]
I did a quick test with ProFTPd and I can see stuff inside square brackets

```
$ rclone ls TestFTP:brackets
        0 with[brackets]yes/inside
```

So this probably depends on which ftp server you are using.  What is it called?

**Comment by azureblaze** [comment]
The server isn't under my control but it is pure-ftpd

**Comment by easy90rider** [comment]
I also did a test:
```
rclone ls ftp:
113844 [test]/SKMBT_.docx
101654729 [test]/SKMBT_.pdf
```

```
rclone ls ftp:Digi\ Cloud/[test]
   113844 SKMBT_C25016030811130.docx
101654729 SKMBT_C25016030811130.pdf
```

**Comment by azureblaze** [comment]
I installed pure-ftpd locally and the issue seems to be with pure-ftpd

```
~ rclone.exe lsd ftp:
2017/05/24 16:25:12 INFO  : localhost: Modify window not supported
           0 2017-03-03 02:25:38        -1 .asciinema
           0 2017-04-20 19:42:15        -1 .config
           0 2017-05-03 07:49:52        -1 .gdrive
           0 2017-03-29 22:49:42        -1 .gnupg
           0 2017-02-27 21:24:44        -1 .local
           0 2017-03-02 01:17:28        -1 .nano
           0 2017-05-23 04:45:33        -1 .ssh
           0 2017-05-24 22:22:22        -1 [test]
           0 2017-05-18 02:10:59        -1 bin
           0 2017-05-24 14:02:40        -1 hashbackup
```
``` 
~ bash.exe -c 'ls /home/*/[test]'
11  2  3252  4  432  5  5532  643
```
but 
```
~ rclone.exe ls ftp:[test] --verbose=5
2017/05/24 16:29:34 DEBUG : rclone: Version "v1.36-120-ga9d29c22" starting with parameters ["C:\\Users\\*\\Documents\\bin\\rclone.exe" "ls" "ftp:[test]" "--verbose=5"]
2017/05/24 16:29:34 DEBUG : localhost: Connecting to FTP server
2017/05/24 16:29:34 INFO  : localhost: Modify window not supported
2017/05/24 16:29:34 DEBUG : Go routines at exit 3
2017/05/24 16:29:34 DEBUG : rclone: Version "v1.36-120-ga9d29c22" finishing with parameters ["C:\\Users\\*\\Documents\\bin\\rclone.exe" "ls" "ftp:[test]" "--verbose=5"]
```

**Comment by ncw** [comment]
Yes this seems to be pureftpd intepreting wildcards.

```
ftp> ls brackets
200 PORT command successful
150 Connecting to port 34975
drwxrwxr-x    3 sftptest   sftptest         4096 May 24 20:51 with[brackets]yes
226-Options: -l 
226 1 matches total
ftp> ls brackets/with[brackets]yes
200 PORT command successful
150 Connecting to port 44283
226-Options: -l 
226 0 matches total
```

Where if we try with wildcards we can see it can see in the directory,

```
ftp> ls brackets/with?brackets?yes
200 PORT command successful
150 Connecting to port 53341
drwxrwxr-x    2 sftptest   sftptest         4096 May 24 20:51 a
-rw-rw-r--    1 sftptest   sftptest            0 May 24 20:50 inside
226-Options: -l 
226 2 matches total
ftp> quit
221-Goodbye. You uploaded 0 and downloaded 0 kbytes.
221 Logout.
```

I wonder if there is some way of turning off wildcard matching...

A quick experiment with curl indicates that it CWD into a directory before listing it, so I could maybe do that.

**Comment by azureblaze** [comment]
If I add `'**'` to the end like `rclone ls ftp:**` it sees the files but then there's the issue that it actually puts `'**' `into the filepath and treats it as the base directory. For example I tried
`~ rclone.exe  copy ftp:** dir` 
and every file had a similar error to the following
`2017/05/31 17:49:15 Failed to copy: failed to open source object: open: 550 Can't open **/100/200/300/100/200/300/536354: No such file or directory`

So basically it seems that special characters are simultaneously being treated as wildcards and part of the literal path.
On a separate note, until now I had assumed filtering was handled locally, is this the case on other remotes or for example does google drive filter using the api when the `--filter` flag is passed to rclone?

**Comment by eephyne** [comment]
I confirm this issue too (way later by the way). I have v1.48.0

**Comment by ncw** [comment]
This bug isn't fixed in pureftpd.

I reported a different bug to the maintainer but he wasn't interested in fixing it and I didn't have the time :-(

https://twitter.com/jedisct1/status/1164513253780799488

Anyone want to fix this in pureftpd?

**Comment by ivandeex** [comment]
@ghost Does the problem reproduce with rclone 1.54?

**Comment by ivandeex** [comment]
This issue is stale since 2019.
I am going to close this ticket because the bug is `outside rclone` and it has been reported to `PureFTP` long ago.
